pkg/controller/options.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/crossplane/crossplane-runtime/pkg/feature"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+)
+
+// DefaultOptions returns a functional set of options with conservative
+// defaults.
+func DefaultOptions() Options {
```

Comment:
@negz is motivation for using a non-pointer type here that we may pass the options to multiple consumers who may in turn desire to modify them? And features being a pointer type ensures that feature flags are enabled consistently throughout all consumers?

Reply:
> Is motivation for using a non-pointer type here that we may pass the options to multiple consumers who may in turn desire to modify them?

Close, but also kind of the opposite. I tend to prefer to pass structs by value when:

1. Consumers are expected to treat the struct's values as read-only.
2. There's no significant performance benefit to passing by reference.

I could see that passing by value might signal "you can mutate your copy", but I feel it also signals "you're not allowed to mutate my copy".

You could make a good argument that `Options` falls under the "small structs that might grow" clause of https://github.com/golang/go/wiki/CodeReviewComments#pass-values, but I'm not super worried about performance loss due to copying this struct given that it's not really in any hot paths.

> And features being a pointer type ensures that feature flags are enabled consistently throughout all consumers?

Actually I think I originally had intended for `*Flags` to be passed by value for similar reasons to `Options`, but the `go vet` `copylocks` check reminded me that I shouldn't be copying mutexes.